frontend/apps/filemanager/filemanagerutil.lua

```diff
@@ -59,8 +63,118 @@ function filemanagerutil.resetDocumentSettings(file)
             end
         end
         doc_settings:makeTrue("docsettings_reset_done") -- for readertypeset block_rendering_mode
-        doc_settings:close()
+        doc_settings:flush()
     end
 end
 
+-- Get a document's status ("new", "reading", "complete", or "abandoned")
+function filemanagerutil.getStatus(file)
+    local status = "new"
+    if DocSettings:hasSidecarFile(file) then
+        local docinfo = DocSettings:open(file) -- no io handles created, do not close
+        if docinfo.data.summary and docinfo.data.summary.status and docinfo.data.summary.status ~= "" then
+            status = docinfo.data.summary.status
+        else
+            status = "reading"
+        end
+    end
+    return status
+end
+
+-- Set a document's status
+function filemanagerutil.setStatus(file, status)
+    -- In case the book doesn't have a sidecar file, this'll create it
+    local docinfo = DocSettings:open(file)
+    local summary
+    if docinfo.data.summary and docinfo.data.summary.status then
+        -- Book already had the full BookStatus table in its sidecar, easy peasy!
+        docinfo.data.summary.status = status
+        docinfo.data.summary.modified = os.date("%Y-%m-%d", os.time())
+        summary = docinfo.data.summary
+    else
+        -- No BookStatus table, create a minimal one...
+        if docinfo.data.summary then
+            -- Err, a summary table with no status entry? Should never happen...
+            summary = { status = status }
+            -- Append the status entry to the existing summary...
+            require("util").tableMerge(docinfo.data.summary, summary)
+            docinfo.data.summary.modified = os.date("%Y-%m-%d", os.time())
+            summary = docinfo.data.summary
+        else
+            -- No summary table at all, create a minimal one
+            summary = {
+                status = status,
+                modified = os.date("%Y-%m-%d", os.time())
+            }
+        end
+    end
+    docinfo:saveSetting("summary", summary)
+    docinfo:flush()
+end
+
+-- Generate a book status file dialog button
+function filemanagerutil.genStatusButton(to_status, current_status, file, caller_callback)
+    local status_text = {
+        reading   = _("Reading"),
+        abandoned = _("On hold"),
+        complete  = _("Finished"),
+    }
+    return {
+        text = status_text[to_status],
+        id = to_status, -- used by covermenu
+        enabled = current_status ~= to_status,
+        callback = function()
+            filemanagerutil.setStatus(file, to_status)
+            caller_callback()
+        end,
+    }
+end
+
+-- Generate all book status file dialog buttons in a row
+function filemanagerutil.getStatusButtonsRow(file, caller_callback)
+    local function genStatusButton(to_status, current_status)
```

Comment:
Since its local now, you can pass `to_status` only.

Reply:
V good